PR: https://github.com/odoo/odoo/pull/80376

From: 78c8965ef62d1b4bcb190bb37a5c0fde2961f418
From: Eleftherios Kritsotakis
Date: 2021-12-09 17:06:09

Structural Changes: 5
Total Changes: 10

[IMP] point_of_sale: more activated settings by default

We provide a better experience when more settings are activated by default.

For this reason, we activated by default the following settings:
1. POS Settings
- Inventory Management is set to "In real time"

2. POS Shop Settings
- Product Prices is set to "Tax-Included Price"
- Product Configurator is activated

task-2616749

closes odoo/odoo#80376

Signed-off-by: Masereel Pierre <pim@odoo.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -72,7 +72,7 @@ class PosConfig(models.Model):
         help='The receipt will automatically be printed at the end of each order.')
     iface_print_skip_screen = fields.Boolean(string='Skip Preview Screen', default=True,
         help='The receipt screen will be skipped if the receipt can be printed automatically.')
-    iface_tax_included = fields.Selection([('subtotal', 'Tax-Excluded Price'), ('total', 'Tax-Included Price')], string="Tax Display", default='subtotal', required=True)
+    iface_tax_included = fields.Selection([('subtotal', 'Tax-Excluded Price'), ('total', 'Tax-Included Price')], string="Tax Display", default='total', required=True)
     iface_start_categ_id = fields.Many2one('pos.category', string='Initial Category',
         help='The point of sale will display this product category by default. If no category is specified, all available products will be shown.')
     iface_available_categ_ids = fields.Many2many('pos.category', string='Available PoS Product Categories',
@@ -140,7 +140,7 @@ class PosConfig(models.Model):
     module_pos_discount = fields.Boolean("Global Discounts")
     module_pos_loyalty = fields.Boolean("Loyalty Program")
     module_pos_mercury = fields.Boolean(string="Integrated Card Payments")
-    product_configurator = fields.Boolean(string="Product Configurator")
+    product_configurator = fields.Boolean(string="Product Configurator", default=True)
     is_posbox = fields.Boolean("PosBox")
     is_header_or_footer = fields.Boolean("Header & Footer")
     module_pos_hr = fields.Boolean(help="Show employee login screen")

--- a/addons/point_of_sale/models/res_company.py
+++ b/addons/point_of_sale/models/res_company.py
@@ -7,9 +7,9 @@ class ResCompany(models.Model):
     _inherit = 'res.company'
 
     point_of_sale_update_stock_quantities = fields.Selection([
-            ('closing', 'At the session closing (recommended)'),
-            ('real', 'In real time'),
-            ], default='closing', string="Update quantities in stock",
+            ('closing', 'At the session closing'),
+            ('real', 'In real time (recommended)'),
+            ], default='real', string="Update quantities in stock",
             help="At the session closing: A picking is created for the entire session when it's closed\n In real time: Each order sent to the server create its own picking")
 
     @api.constrains('period_lock_date', 'fiscalyear_lock_date')
